homeassistant/components/tradfri/base_class.py

```diff
@@ -122,8 +123,19 @@ def __init__(
     ) -> None:
         """Initialize a device."""
         self._attr_available = device.reachable
+        self._hub_available = True
         super().__init__(device, api, gateway_id)
 
+    def set_hub_available(self, available: bool) -> None:
+        """Set status of hub."""
+        if available != self._hub_available:
+            self._hub_available = available
+            self._refresh(self._device)
+
+    def setup(self) -> None:
+        """Connect signal handler."""
+        async_dispatcher_connect(self.hass, SIGNAL_GW, self.set_hub_available)
```

Comment:
Please move this to `async_added_to_hass`.

Reply:
Done.